store/displayPreferencesApi.ts

```diff
@@ -0,0 +1,241 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
+import { MutationTree, ActionTree } from 'vuex';
+import { toString as toStr } from 'lodash';
+import destr from 'destr';
+import { defaultState as settingState } from '~/store/settings';
+
+/**
+ * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+export interface CustomPreferences {
+  darkMode: boolean;
+  locale: string;
+}
+/**
+ * DisplayPreferencesDto but with our custom typings for CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientPreferences
+  extends Omit<DisplayPreferencesDto, 'CustomPrefs'> {
+  CustomPrefs?: CustomPreferences;
+}
+
+export interface DisplayPreferencesApiState {
+  syncing: boolean;
+  LastSync: number;
+  LastSettingChange: number;
+}
+
+const defaultState = (): DisplayPreferencesApiState => ({
+  syncing: false,
+  LastSync: -1,
+  LastSettingChange: -1
+});
+
+/**
+ * Cast custom preferences returned from the server from strings to the correct Javascript type
+ *
+ * @param {DisplayPreferencesDto} data - Response from the server
+ * @returns {ClientPreferences} - The response with the correct datatypes.
+ */
+function castResponse(data: DisplayPreferencesDto): ClientPreferences {
+  const result = data as ClientPreferences;
+  for (const [key, value] of Object.entries(
+    result.CustomPrefs as CustomPreferences
+  )) {
+    // @ts-expect-error - TypeScript can't infer types from Object.entries
+    const stateValue = settingState()[key];
+    if (typeof stateValue === 'boolean') {
+      // @ts-expect-error - TypeScript can't infer types from Object.entries
+      result.CustomPrefs?.[key] = Boolean(value);
+    } else if (typeof stateValue === 'number') {
+      // @ts-expect-error - TypeScript can't infer types from Object.entries
+      result.CustomPrefs?.[key] = Number(value);
+    } else if (typeof stateValue === 'object') {
+      if (Array.isArray(stateValue)) {
```

Comment:
In that case, if our server value is supposed to be an array, wouldn't it be stored as a JSON that we could directly parse?